src/coreComponents/mesh/generators/CellBlockManager.hpp

```diff
@@ -0,0 +1,234 @@
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CellBlockManager.hpp
+ */
+
+#ifndef GEOSX_MESH_CELLBLOCKMANAGER_H_
+#define GEOSX_MESH_CELLBLOCKMANAGER_H_
+
+#include "mesh/generators/CellBlockManagerABC.hpp"
+#include "mesh/generators/CellBlock.hpp"
+
+namespace geosx
+{
+
+/**
+ * @class CellBlockManager
+ * @brief The CellBlockManager class provides an interface to ObjectManagerBase in order to manage CellBlock data.
+ */
+class CellBlockManager : public CellBlockManagerABC
+{
+public:
+
+  /**
+   * @brief Constructor for CellBlockManager object.
+   * @param name name of this instantiation of CellBlockManager
+   * @param parent pointer to the parent Group of this instantiation of CellBlockManager
+   */
+  CellBlockManager( string const & name, Group * const parent );
+
+  CellBlockManager( const CellBlockManager & ) = delete;
+
+  CellBlockManager & operator=( const CellBlockManager & ) = delete;
+
+  ~CellBlockManager() override = default;
+
+  virtual Group * createChild( string const & childKey, string const & childName ) override;
+
+  /**
+   * @brief Maximum number of faces allowed (in memory) per each node.
+   * @return The number as an integer.
+   */
+  static constexpr int maxFacesPerNode()
+  { return 200; }
+
+  /**
+   * @brief Extra space for node to faces mapping.
+   * @return Number of extra values as an integer.
+   */
+  static constexpr localIndex getFaceMapOverallocation()
+  { return 8; }
+
+  array2d< real64, nodes::REFERENCE_POSITION_PERM > const & getNodesPositions() const override;
+
+  /**
+   * @brief Returns a mutable reference to vector holding the nodes coordinates
+   * @return The reference
+   *
+   * @note This is meant to be used as a setter. Do not resize the vector yourself.
+   */
+  array2d< real64, nodes::REFERENCE_POSITION_PERM > & getNodesPositions();
+
+  ArrayOfSets< localIndex > getNodeToEdges() const override;
+
+  ArrayOfSets< localIndex > getNodeToFaces() const override;
+
+  ArrayOfArrays< localIndex > getNodeToElements() const override;
+
+  array2d< geosx::localIndex > const & getEdgeToNodes() const override;
+
+  ArrayOfSets< geosx::localIndex > const & getEdgeToFaces() const override;
+
+  ArrayOfArrays< localIndex > getFaceToNodes() const override;
+
+  ArrayOfArrays< geosx::localIndex > const & getFaceToEdges() const override;
+
+  array2d< localIndex > getFaceToElements() const override;
+
+  array1d< globalIndex > getNodeLocalToGlobal() const override;
+
+  /**
+   * @brief Returns a mutable reference to vector holding the node to global mapping.
+   * @return The reference
+   *
+   * @note This is meant to be used as a setter. Do not resize the vector yourself.
+   */
+  array1d< globalIndex > & getNodeLocalToGlobal();
```

Comment:
Should we do something to prevent resizing? Like returning an `arrayView1d` instead of the reference to the `array1d`?

Reply:
An `array1d< globalIndex > const &` would do the trick too. But I feel like we probably shouldn't be modifying this outside of the class anyways and certainly 95% of the use cases are read only so I'd be inclined to just make it an `arrayView1d< globalIndex const >`.

Reply:
For this precise case, the original code from `ObjectManagerBase` does
```c++
  arrayView1d< globalIndex > localToGlobalMap()
  { return m_localToGlobalMap; }
```
The pattern (from `InternalMeshGenerator` and `PAMELAMeshGenerator`) is to fill it through `arrayView1d< globalIndex > const &` which I can reproduce.

But more important is the API at `CellBlockManagerABC`. This is actually the second tick box of this PR description (still unticked 😉 ): "_The way to return the arrays must be improved (I’ll need help here)._".

My way of returning the arrays is quite inconsistent, and therefore surely incorrect at some point.

I think (correct me please) that simply returning a `array1d< globalIndex >` allows `move` semantics which was a nice effect (while far from being mandatory, specially if we deallocate the `CellBlockManager` instance). But what about `const` correctness?